src/aap_eda/services/activation/engine/kubernetes.py

```diff
@@ -133,20 +132,21 @@ def start(self, request: ContainerRequest, log_handler: LogHandler) -> str:
             self.cleanup(self.job_name, log_handler)
             raise
 
-    def get_status(self, container_id: str) -> ActivationStatus:
+    def get_status(self, container_id: str) -> tuple[ActivationStatus, str]:
         pod = self._get_job_pod(container_id)
 
+        message = ""
         container_status = pod.status.container_statuses[0]
         if container_status.state.running:
             status = ActivationStatus.RUNNING
         elif container_status.state.terminated:
             exit_code = container_status.state.terminated.exit_code
-            # should change 143 to ActivationStatus.STOPPED?
-            if exit_code in SUCCESSFUL_EXIT_CODES:
+            if exit_code == 0:
                 status = ActivationStatus.COMPLETED
                 LOGGER.info("Pod has successfully exited")
             else:
                 status = ActivationStatus.FAILED
+                message = container_status.state.terminated.message
```

Comment:
Hi @mkanoor @hsong-rh 
Here is an example of my concerns. 
Here we lose the control over the content of the message potentially returning to the user unclear or complex technical messages. I really tend to refuse to use this in the status_message since this field is exposed to the final user. 
In this case, we should take the exit code and return and well known message like "Container has failed with {exit_code} code" and that message should be placed in the messages.py file and used also by podman to have a consistent format in both engines. The specific message returned from the engine should be logged to allow the user or administrator to get more details for debugging.

Reply:
@Alex-Izquierdo Our software is not meant for business/home users most of our users are going to be Administrators/Operators who would need more information and they would be able to make sense of the message. Plus I think we are expecting podman/k8s to put a meaningful message in there I hope they don't put a stack trace. For now as we are developing it's painful to scroll thru logs to get the error message. For now let us keep it this way and we can bring it up in dev sync and discuss

Reply:
ok, I buy. But lets use a dataclass instead of tuple, please.

Reply:
For all not known messages, we should be consistent and reuse them between engines, like `"Pod {container_status} is not running or terminated"`